client/my-sites/plans/plans-features/google-analytics.js

```diff
@@ -33,7 +33,12 @@ export default function( context ) {
 
 	function goBack() {
-		page.back( '/plans/' + ( context.params.domain || '' ) );
+		const fallback = '/plans/' + ( context.params.domain || '' );
+		if ( ! context.prevPath || context.prevPath.indexOf( context.params.domain ) === -1 ) {
```

Comment:
Yes, it does, but only when history is empty.
But in case we come from site-select, we dont really want to go back there.
We want to /plans, that's @mtias stance on it and i kinda agree :)